src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -40,8 +46,40 @@ case class AvroRelation(location: String)(@transient val sqlContext: SQLContext)
     convertedSchema
   }
 
+  // There are several properties of Avro that we have to account for in this method:
+  // 1) Avro uses Utf8 strings, so we want to convert them to java.lang.String for SparkSQL,
+  //    including making the appropriate recursive calls.
+  // 2) Avro map keys are always strings, so we don't need to recurse on them when processing maps.
+  // 3) Byte arrays and ByteBuffers are reused by Avro, so we must copy them out.
+  def convertToSparkSQL(obj: Any): Any = {
+    obj match {
+      case u: Utf8 => u.toString
+      case m: Map[Any, Any] => m.map(x => (x._1.toString, convertToSparkSQL(x._2)))
+      case a: GenericData.Array[Any] => {
+        val ar = new Array[Any](a.size)
+        var idx = 0
+        while (idx < a.size) {
+          ar(idx) = convertToSparkSQL(a(0))
```

Comment:
should this be a(idx)? Also, consider more descriptive names (something like a -> avroArray; ar -> javaArray)